src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -579,17 +579,74 @@ def process_declarations(self, parent, nodes, arg_list):
+                   and not parent.symbol_table.imported_symbols(container):
+                    # TODO #11 Log the fact that this explicit symbol import
+                    # will replace a previous import with an empty only-list.
+                    pass
                 for name in decl.items[4].items:
-                    parent.symbol_table.add(
-                        DataSymbol(str(name).lower(),
-                                   datatype=DataType.DEFERRED,
-                                   interface=GlobalInterface(container)))
+                    # The DataSymbol adds itself to the list of symbols
+                    # imported by the Container referenced in the
+                    # GlobalInterface.
+                    sym_name = str(name).lower()
+                    if sym_name not in parent.symbol_table:
+                        parent.symbol_table.add(
+                            DataSymbol(sym_name,
+                                       datatype=DataType.DEFERRED,
+                                       interface=GlobalInterface(container)))
+                    else:
+                        # There's already a symbol with this name
+                        existing_symbol = parent.symbol_table.lookup(sym_name)
+                        if not existing_symbol.is_global:
+                            raise SymbolError(
+                                "Symbol '{0}' is imported from module '{1}' "
+                                "but is already present in the symbol table as"
+                                " either an argument or a local ({2}).".
+                                format(sym_name, mod_name,
+                                       str(existing_symbol)))
+                        # TODO #11 Log the fact that we've already got an
+                        # import of this symbol and that will take precendence.
+            elif not decl.items[3]:
+                # We have a USE statement without an ONLY clause.
+                if (not new_container) and (not container.wildcard_import) \
+                   and (not parent.symbol_table.imported_symbols(container)):
+                    # TODO #11 Log the fact that this explicit symbol import
+                    # will replace a previous import with an empty only-list.
+                    pass
+                container.wildcard_import = True
+            elif decl.items[3].lower().replace(" ", "") == ",only:":
+                # This use has an 'only: ' but no associated list of
+                # imported symbols. (It serves to keep a module in scope while
+                # not actually importing anything from it.)
+                if not new_container and \
+                   (container.wildcard_import or
+                    parent.symbol_table.imported_symbols(container)):
+                    # TODO #11 Log the fact that this import with an empty
+                    # only-list is irrelevant because of existing 'use's of
```

Comment:
Perhaps change `irrelevant` to `ignored` as an existing container remains unchanged if `only:` is found.

Reply:
Done.